agent/discover/discover.go

```diff
@@ -0,0 +1,52 @@
+package discover
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/codeskyblue/go-sh"
+	consul "github.com/hashicorp/consul/api"
+)
+
+type Discover func(client consul.Client) DiscoverInterface
+
+type discoverFact struct {
+	cStorePath       string
+	discoveryCommand []string
+}
+
+type DiscoverInterface interface {
+	shouldDiscover(client consul.Client) bool
+	KVStorePrefix() string
+}
+
+func StoreDiscovery(discover DiscoverInterface, client consul.Client, cStorePath, cStoreValue string) {
+	kvPath := fmt.Sprintf("%s/%s", discover.KVStorePrefix(), cStorePath)
+
+	log.Print("kvPath", kvPath)
+
+	client.KV().Put(&consul.KVPair{
+		Key:   kvPath,
+		Value: []byte(cStoreValue)}, nil)
+}
+
+func CollectDiscoveries(discover DiscoverInterface, clusterDiscoveries []discoverFact, client consul.Client) error {
+	discover_session := sh.NewSession()
```

Comment:
What is this `sh.NewSession` for? To understand the logic

Reply:
Comment added. this code is right now not used. the idea is that we can use a declarative syntax (as an array initially, but maybe a yaml configuration file later on) with "shell command to run" and whee the stdout of the shell command is directly recorded as a value in the given KVStore path. 

I originally thought we'll need this, but then realized that the current JIRA ticket does not require this. I just wanted to keep it in because we'll most likely need it at some later point in time. 

If its confusing I can remove it for now and store it in a separate branch.

Reply:
I removed it to reduce the review obstacle.